### PR TITLE
Implement embeddable dashboard on Malware Detection module

### LINKED ISSUE #6520: [Remove discover] Implement embeddable dashboard on Malware Detection module
## Description

We have to implement the embeddable dashboard on Malware Detection -> dashboard tab and deprecate any use of _kibana-integrations_ components.

## Tasks

- [x] Implement the embeddable dashboard on Malware Detection -> dashboard tab

### PR DESCRIPTION
**Pull request by yenienserrano**
### Description
This pull request replaces the `Dashboard` tab on Malware Detection application.

### Issues Resolved
#6520 

### Evidence

image
image

### Test

# Global

| Test | Chrome | Firefox | Safari |
| --- |  --- | --- | --- |
| Check data results consistency between events and dashboard tab | :black_circle: | :black_circle: | :black_circle: |
| Check the filters persist between the events and dashboard tab | :black_circle: | :black_circle: | :black_circle: |
| Check the filters are cleaned when throwing in the events tab | :black_circle: | :black_circle: | :black_circle: |
| Check cluster-manager filter on tabs. Must be applied when the tab is shown (all tabs) | :black_circle: | :black_circle: | :black_circle: |
| Check pinned agent filter on tabs | :black_circle: | :black_circle: | :black_circle: |
| Check the request with hidden filters applied (allow agents, hide manager alerts)  | :black_circle: | :black_circle: | :black_circle: |
| Check the fixer filter cannot be removed from the filter bar (the remove icon is hidden) | :black_circle: | :black_circle: | :black_circle: |

- Check data results consistency between events and dashboard tab

<details>

</details>

-  Check the filters persist between the events and dashboard tab

<details>

</details>

- Check the filters are cleaned when throwing in the events tab

<details>

</details>

- Check cluster-manager filter on tabs. Must be applied when the tab is shown (all tabs)

<details>

</details>

- Check pinned agent filter on tabs

<details>

</details>

- Check the request with hidden filters applied (allow agents, hide manager alerts)

<details>

`Enable/disable the hideManagerAlerts settings on App Settings`

Screenshot 2024-03-31 at 12 29 47

</details>

- Check the fixed filter cannot be removed from the filter bar

<details>

</details>

# Dashboard tab

| Test | Chrome | Firefox | Safari |
| --- |  --- | --- | --- |
| Use search bar input to filter any word and check if the dashboard shows the corresponding results | :black_circle: | :black_circle: | :black_circle: |
| Select an option from one of the recommended filters and check that the corresponding filter is added and the dashboard changes according to the applied filter | :black_circle: | :black_circle: | :black_circle: |
| Add a filter through interaction with a visualization and check that the dashboard updates | :black_circle: | :black_circle: | :black_circle: |
| Use the input and type an incorrect word to check if the no results message is shown | :black_circle: | :black_circle: | :black_circle: |

- Use search bar input to filter any word and check if the dashboard shows the corresponding results

<details>

</details>

- Select an option from one of the recommended filters and check that the corresponding filter is added and the dashboard changes according to the applied filter

<details>

</details>

- Add a filter through interaction with a visualization and check that the dashboard updates

<details>

</details>

- Use the input and type an incorrect word to check if the no results message is shown

<details>

</details>

# Events tab (Discover)

## Search bar

| Test | Chrome | Firefox | Safari |
| --- |  --- | --- | --- |
| Apply filters using search bar input | :black_circle: | :black_circle: | :black_circle: |
| Apply filters using the search bar `add filter` feature | :black_circle: | :black_circle: | :black_circle: |
| Apply filters using the date picker | :black_circle: | :black_circle: | :black_circle: |
| Apply filters with no results (Check the `No results match your search criteria` message)| :black_circle: | :black_circle: | :black_circle: |
| Use the `saved queries` feature (Check saved queries CRUD) | :black_circle: | :black_circle: | :black_circle: |

- Apply filters using search bar input

<details>

</details>

- Apply filters using the search bar `add filter` feature

<details>

</details>

- Apply filters using the date picker

<details>

</details>

- Apply filters with no results (Check the `No results match your search criteria` message)

<details>

</details>

- Use the `saved queries` feature (Check saved queries CRUD)

<details>

</details>

## Histogram chart

| Test | Chrome | Firefox | Safari |
| --- |  --- | --- | --- |
| Check histogram chart hits consistency | :black_circle: | :black_circle: | :black_circle: |
| Use chart options (three points: maximize, minimize, inspect) | :black_circle: | :black_circle: | :black_circle: |

- Check histogram chart hits consistency 

<details>

</details>

- Use chart options (three points: maximize, minimize, inspect) 

<details>

</details>

## Data grid

| Test | Chrome | Firefox | Safari |
| --- |  --- | --- | --- |
| Check all table header features (sort, show/hide columns, density, etc) | :black_circle: | :black_circle: | :black_circle: |
| Check table pagination and rows per page | :black_circle: | :black_circle: | :black_circle: |
| Check document detail flyout | :black_circle: | :black_circle: | :black_circle: |

- Check all table header features (sort, show/hide columns, density, etc)

<details>

</details>

- Check table pagination and rows per page

<details>

</details>

- Check document detail flyout

<details>

</details>

### Check List
- [x] All tests pass
  - [x] `yarn test:jest`
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [ ] Update [CHANGELOG.md](./../CHANGELOG.md)
- [x] Commits are signed per the DCO using --signoff

### CODE REVIEW

File: plugins/main/public/components/overview/malware-detection/dashboard/dashboard.tsx [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+                showDatePicker={true}
+                showQueryInput={true}
+                showQueryBar={true}
+              />
+            </div>
+          )}
+          <SampleDataWarning />
+          {dataSource && results?.hits?.total === 0 ? (
+            <DiscoverNoResults />
+          ) : null}
```
Comment (JuanGarriuz) [inline plugins/main/public/components/overview/malware-detection/dashboard/dashboard.tsx:103]: If the sample data warning were added in the rendering of the condition to display the no results, the warning would always be shown as long as there are results, thus avoiding showing the sample data while no results are visible.

4.9.0 Dashboard

image

Embeddable

image

Comment (yenienserrano) [reply plugins/main/public/components/overview/malware-detection/dashboard/dashboard.tsx]: Resolved in the commit [94cd1a2](https://github.com/wazuh/wazuh-dashboard-plugins/pull/6521/commits/94cd1a26cf6c031e19d43b9a8839cf5fd471cbc7)

**Comment by lucianogorza** [comment]
### Test

# Global

| Test | Chrome | Firefox | Safari |
| --- |  --- | --- | --- |
| Check data results consistency between events and dashboard tab | :black_circle: | :black_circle: | :black_circle: |
| Check the filters persist between the events and dashboard tab | :black_circle: | :black_circle: | :black_circle: |
| Check the filters are cleaned when throwing in the events tab | :black_circle: | :black_circle: | :black_circle: |
| Check cluster-manager filter on tabs. Must be applied when the tab is shown (all tabs) | :black_circle: | :black_circle: | :black_circle: |
| Check pinned agent filter on tabs | :black_circle: | :black_circle: | :black_circle: |
| Check the request with hidden filters applied (allow agents, hide manager alerts)  | :black_circle: | :black_circle: | :black_circle: |
| Check the fixer filter cannot be removed from the filter bar (the remove icon is hidden) | :black_circle: | :black_circle: | :black_circle: |

- Check data results consistency between events and dashboard tab

<details>


</details>

-  Check the filters persist between the events and dashboard tab

<details>


</details>

- Check the filters are cleaned when throwing in the events tab

<details>


</details>

- Check cluster-manager filter on tabs. Must be applied when the tab is shown (all tabs)

<details>


</details>

- Check pinned agent filter on tabs

<details>


</details>

- Check the request with hidden filters applied (allow agents, hide manager alerts)

<details>

`Enable/disable the hideManagerAlerts settings on App Settings`

Screenshot 2024-03-31 at 12 29 47


</details>

- Check the fixed filter cannot be removed from the filter bar

<details>

</details>


# Dashboard tab

| Test | Chrome | Firefox | Safari |
| --- |  --- | --- | --- |
| Use search bar input to filter any word and check if the dashboard shows the corresponding results | :black_circle: | :black_circle: | :black_circle: |
| Select an option from one of the recommended filters and check that the corresponding filter is added and the dashboard changes according to the applied filter | :black_circle: | :black_circle: | :black_circle: |
| Add a filter through interaction with a visualization and check that the dashboard updates | :black_circle: | :black_circle: | :black_circle: |
| Use the input and type an incorrect word to check if the no results message is shown | :black_circle: | :black_circle: | :black_circle: |

- Use search bar input to filter any word and check if the dashboard shows the corresponding results

<details>


</details>

- Select an option from one of the recommended filters and check that the corresponding filter is added and the dashboard changes according to the applied filter

<details>


</details>

- Add a filter through interaction with a visualization and check that the dashboard updates

<details>


</details>

- Use the input and type an incorrect word to check if the no results message is shown

<details>


</details>


# Events tab (Discover)

## Search bar

| Test | Chrome | Firefox | Safari |
| --- |  --- | --- | --- |
| Apply filters using search bar input | :black_circle: | :black_circle: | :black_circle: |
| Apply filters using the search bar `add filter` feature | :black_circle: | :black_circle: | :black_circle: |
| Apply filters using the date picker | :black_circle: | :black_circle: | :black_circle: |
| Apply filters with no results (Check the `No results match your search criteria` message)| :black_circle: | :black_circle: | :black_circle: |
| Use the `saved queries` feature (Check saved queries CRUD) | :black_circle: | :black_circle: | :black_circle: |

- Apply filters using search bar input

<details>


</details>

- Apply filters using the search bar `add filter` feature

<details>


</details>

- Apply filters using the date picker

<details>


</details>

- Apply filters with no results (Check the `No results match your search criteria` message)

<details>


</details>

- Use the `saved queries` feature (Check saved queries CRUD)

<details>


</details>


## Histogram chart

| Test | Chrome | Firefox | Safari |
| --- |  --- | --- | --- |
| Check histogram chart hits consistency | :black_circle: | :black_circle: | :black_circle: |
| Use chart options (three points: maximize, minimize, inspect) | :black_circle: | :black_circle: | :black_circle: |

- Check histogram chart hits consistency 

<details>


</details>


- Use chart options (three points: maximize, minimize, inspect) 

<details>


</details>

## Data grid

| Test | Chrome | Firefox | Safari |
| --- |  --- | --- | --- |
| Check all table header features (sort, show/hide columns, density, etc) | :black_circle: | :black_circle: | :black_circle: |
| Check table pagination and rows per page | :black_circle: | :black_circle: | :black_circle: |
| Check document detail flyout | :black_circle: | :black_circle: | :black_circle: |

- Check all table header features (sort, show/hide columns, density, etc)

<details>


</details>

- Check table pagination and rows per page

<details>


</details>


- Check document detail flyout

<details>


</details>

**Comment by lucianogorza** [comment]
### Inconsistent chart and axis titles
image

**Review by lucianogorza — CHANGES_REQUESTED** [review]
- In the Dashboard tab, the table shows only 4 records, while the Inventory tab has 357.
- The fixed filters can be removed after refresh the entire page.

Details in comment https://github.com/wazuh/wazuh-dashboard-plugins/pull/6521#issuecomment-2083049894

**Review by lucianogorza — APPROVED** [review]
## CR 🟢 
## Test 🟢 

### Consistent labels and chart types
image

The other warnings in the comment https://github.com/wazuh/wazuh-dashboard-plugins/pull/6521#issuecomment-2083049894 are not responsibilities of this PR.

**Comment by JuanGarriuz** [comment]
### Test

# Global

| Test | Chrome | Firefox | Safari |
| --- |  --- | --- | --- |
| Check data results consistency between events and dashboard tab | :green_circle: | :black_circle: | :black_circle: |
| Check the filters persist between the events and dashboard tab | :green_circle: | :black_circle: | :black_circle: |
| Check the filters are cleaned when throwing in the events tab | :green_circle: | :black_circle: | :black_circle: |
| Check cluster-manager filter on tabs. Must be applied when the tab is shown (all tabs) | :green_circle: | :black_circle: | :black_circle: |
| Check pinned agent filter on tabs | :green_circle: | :black_circle: | :black_circle: |
| Check the request with hidden filters applied (allow agents, hide manager alerts)  | :green_circle: | :black_circle: | :black_circle: |
| Check the fixer filter cannot be removed from the filter bar (the remove icon is hidden) | :green_circle: | :black_circle: | :black_circle: |

- Check data results consistency between events and dashboard tab

<details>

image
image

</details>

-  Check the filters persist between the events and dashboard tab

<details>

image
image

</details>

- Check the filters are cleaned when throwing in the events tab

<details>

image

</details>

- Check cluster-manager filter on tabs. Must be applied when the tab is shown (all tabs)

<details>

image

</details>

- Check pinned agent filter on tabs

<details>

image

</details>

- Check the request with hidden filters applied (allow agents, hide manager alerts)

<details>

image
image


</details>

- Check the fixed filter cannot be removed from the filter bar

<details>

image

</details>


# Dashboard tab

| Test | Chrome | Firefox | Safari |
| --- |  --- | --- | --- |
| Use search bar input to filter any word and check if the dashboard shows the corresponding results | :green_circle: | :black_circle: | :black_circle: |
| Select an option from one of the recommended filters and check that the corresponding filter is added and the dashboard changes according to the applied filter | :green_circle: | :black_circle: | :black_circle: |
| Add a filter through interaction with a visualization and check that the dashboard updates | :green_circle: | :black_circle: | :black_circle: |
| Use the input and type an incorrect word to check if the no results message is shown | :green_circle: | :black_circle: | :black_circle: |

- Use search bar input to filter any word and check if the dashboard shows the corresponding results

<details>

image

</details>

- Select an option from one of the recommended filters and check that the corresponding filter is added and the dashboard changes according to the applied filter

<details>

image

</details>

- Add a filter through interaction with a visualization and check that the dashboard updates

<details>

image
image

</details>

- Use the input and type an incorrect word to check if the no results message is shown

<details>

image

</details>


# Events tab (Discover)

## Search bar

| Test | Chrome | Firefox | Safari |
| --- |  --- | --- | --- |
| Apply filters using search bar input | :green_circle: | :black_circle: | :black_circle: |
| Apply filters using the search bar `add filter` feature | :green_circle: | :black_circle: | :black_circle: |
| Apply filters using the date picker | :green_circle: | :black_circle: | :black_circle: |
| Apply filters with no results (Check the `No results match your search criteria` message)| :green_circle: | :black_circle: | :black_circle: |
| Use the `saved queries` feature (Check saved queries CRUD) | :green_circle: | :black_circle: | :black_circle: |

- Apply filters using search bar input

<details>

image

</details>

- Apply filters using the search bar `add filter` feature

<details>

image

</details>

- Apply filters using the date picker

<details>

image

</details>

- Apply filters with no results (Check the `No results match your search criteria` message)

<details>

image

</details>

- Use the `saved queries` feature (Check saved queries CRUD)

<details>

image

</details>


## Histogram chart

| Test | Chrome | Firefox | Safari |
| --- |  --- | --- | --- |
| Check histogram chart hits consistency | :green_circle: | :black_circle: | :black_circle: |
| Use chart options (three points: maximize, minimize, inspect) | :green_circle: | :black_circle: | :black_circle: |

- Check histogram chart hits consistency 

<details>

image

</details>


- Use chart options (three points: maximize, minimize, inspect) 

<details>


https://github.com/wazuh/wazuh-dashboard-plugins/assets/124377319/50df0106-eb8a-48c6-86ef-131ccb37427c


</details>

## Data grid

| Test | Chrome | Firefox | Safari |
| --- |  --- | --- | --- |
| Check all table header features (sort, show/hide columns, density, etc) | :green_circle: | :black_circle: | :black_circle: |
| Check table pagination and rows per page | :green_circle: | :black_circle: | :black_circle: |
| Check document detail flyout | :green_circle: | :black_circle: | :black_circle: |

- Check all table header features (sort, show/hide columns, density, etc)

<details>


https://github.com/wazuh/wazuh-dashboard-plugins/assets/124377319/daaeeaeb-8050-4fad-ba6a-e4ca03b18ff1

</details>


- Check document detail flyout

<details>

image

</details>

**Review by JuanGarriuz — APPROVED** [review]
LGTM!!

**Pull request merged**